cdap-data-fabric/src/main/java/co/cask/cdap/spi/metadata/MetadataDirective.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.metadata;
+
+/**
+ * A directive instructing specific behavior when updating metadata.
+ */
+public enum MetadataDirective {
```

Comment:
Just wondering whats the use case of this, when is the time that we want to keep or preserve the metadata if someone changes it.

Reply:
When a entity is updated, we currently have two special handlings:
- the creationTime never gets updated (PRESERVE)
- if the new entity has no description, but there is an existing description, we keep it. (KEEP)

I am trying to preserve that logic

Reply:
Got it, thanks!